File: careconnect2025/backend/core/src/main/java/com/careconnect/model/PackageItem.java
package com.careconnect.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;
import java.time.OffsetDateTime;

@Data @NoArgsConstructor @AllArgsConstructor @Builder
public class PackageItem {
    private String trackingNumber;

    @JsonProperty("expectedDateIso")
    private OffsetDateTime expectedDeliveryDate;  // null if unknown

    @JsonProperty("actions")
    private ActionLinks actionLinks;
}

File: careconnect2025/frontend/lib/features/usps/domain/models/package_item.dart
class PackageItem {
  final String trackingNumber;
  final String? sender;
  final String? expectedDateIso;
  final ActionLinks actions;

  const PackageItem({
    required this.trackingNumber,
    this.sender,
    this.expectedDateIso,
    required this.actions,
  });
}
